src/Kiota.Builder/Writers/Java/JavaConventionService.cs

```diff
@@ -119,7 +119,7 @@ internal void AddRequestBuilderBody(CodeClass parentClass, string returnType, La
     {
         var pathParametersProperty = parentClass.GetPropertyOfKind(CodePropertyKind.PathParameters);
         var requestAdapterProp = parentClass.GetPropertyOfKind(CodePropertyKind.RequestAdapter);
-        var urlTemplateParams = urlTemplateVarName ?? pathParametersProperty?.Name;
+        var urlTemplateParams = string.IsNullOrEmpty(urlTemplateVarName) ? pathParametersProperty?.Name : urlTemplateVarName;
```

Comment:
what about the case where `pathParametersProperty?.Name` return null or empty?

Reply:
we'll technically generate code that doesn't build, but at least kiota itself won't throw (string interpolation considers null values as empty strings, not like TS where you end up with undefined)

Reply:
Not ideal, but I guess there's no other options available to us at this point?

Reply:
since we enabled NRT and treat warnings as errors (which overall is a good thing) we now have string empty where we used to have nulls. Which has a downside on the null coalesce operator. I'm not aware of a version of null coal that also checks for string emptiness.